### DISCUSSION TITLE
Execute existing cronjob on demand

### DISCUSSION THREAD

**Discussion by iriza841** [kubernetes-client/csharp#1578] [Q&A] [answered]
Hello,

I need the ability to execute a cronjob on demand if needed, I found that can be done with next command in kubernetes:
`kubectl create job --from=cronjob/<name of cronjob> <name of job>`

Is this the right way to do it with kubernetesClient?
```
var kubernetesNamespace = "default";
V1CronJob cronJob = await client.ReadNamespacedCronJobAsync("cronJobName", kubernetesNamespace);

var apiVersion = cronJob.ApiVersion;
var metadata = cronJob.Metadata;
metadata.Name = $"{metadata.Name}-manual-{DateTime.Now.ToString("ddMMyyHHmmss")}";

var spec = new V1JobSpec() 
{
    Template = cronJob.Spec.JobTemplate.Spec.Template
};

var job = new V1Job() {
    ApiVersion = apiVersion,
    Kind = V1Job.KubeKind,
    Metadata = metadata, 
    Spec = spec
};

job = await client.BatchV1.CreateNamespacedJobAsync(job, kubernetesNamespace);
```

Thanks in advance,
Regards

**Comment by tg123** [comment, marked as answer]
your code looks good to me
any problem with it?

**Reply by iriza841** [reply]
No, In fact I don't know how to test it yet, I was just asking if it's the equivalent of this:
`kubectl create job --from=cronjob/<name of cronjob> <name of job>`

Thank very much for your answer.
Regards
